spmd/tensor/redistribute.py

```diff
@@ -10,9 +10,9 @@
 
 def redistribute_spmd_tensor(input, device_mesh, placements):
     current_placements = input.placements
+    assert len(placements) == 1, "Only support 1-d placement for now"
+    assert input.device_mesh.mesh.equal(device_mesh.mesh), "cross mesh comm not support yet"
     if isinstance(current_placements[0], Shard) and isinstance(placements[0], Replicate):
```

Comment:
Do we want to support `_Partial` -> `Shard` as well?

Reply:
Just added this :)